docs/error_codes.md

```diff
@@ -1,49 +1,113 @@
 ## Error codes
 
 ### General init related
-- `GS000`: `Could not finish initialization`
-- `GS001`: `Threshold needs to be defined`
-- `GS002`: `A call to set up modules couldn't be executed because the destination account was not a contract`
+- `GS000`: Could not finish initialization
+  - **Why:** Delegatecall to a module during initialization failed (see `setupModules`).
+  - **How to debug/solve:** Ensure the module address is correct and the called contract does not revert.
+- `GS001`: Threshold needs to be defined
+  - **Why:** The signature threshold is not set (equals 0) when checking signatures.
+  - **How to debug/solve:** Make sure the Safe is properly initialized and the threshold is greater than 0.
+- `GS002`: A call to set up modules couldn't be executed because the destination account was not a contract
+  - **Why:** Attempt to initialize a module at an address that is not a contract.
+  - **How to debug/solve:** Check that the module address points to a deployed contract.
 
-### General gas/ execution related
-- `GS010`: `Not enough gas to execute Safe transaction`
-- `GS011`: `Could not pay gas costs with ether`
-- `GS012`: `Could not pay gas costs with token`
-- `GS013`: `Safe transaction failed when gasPrice and safeTxGas were 0` (Deprecated in `v1.5.0`)
+### General gas/execution related
+- `GS010`: Not enough gas to execute Safe transaction
+  - **Why:** `execTransaction` was called with insufficient gas.
+  - **How to debug/solve:** Increase the gas limit for the transaction.
+- `GS011`: Could not pay gas costs with ether
+  - **Why:** Payment for gas in ETH failed (e.g., recipient cannot receive ETH).
+  - **How to debug/solve:** Ensure the receiver address is correct and can accept ETH.
+- `GS012`: Could not pay gas costs with token
+  - **Why:** Payment for gas in token failed (e.g., token does not support transfer or returns false).
+  - **How to debug/solve:** Check that the token supports the transfer standard and the Safe has enough balance.
+- `GS013`: Safe transaction failed when gasPrice and safeTxGas were 0 (Deprecated in v1.5.0)
+  - **Why:** Migration or similar function requiring delegatecall was called outside of delegatecall context or with invalid parameters.
+  - **How to debug/solve:** Only use supported migration scenarios and ensure the call is made via delegatecall.
```

Comment:
This happens on general transaction revert, and is not specific to delegatecall.